### PR TITLE
Work around CPUs offline issue for Kata

### PR DESCRIPTION
**Pull request by RobertKrawitz**
Kata in 4.9 and 4.10 has a problem with hotplugging CPUs whereby CPUs added by means of a CPU resource limit don't all get onlined.  This workaround is to use a pod annotation to change the default_vcpus setting to boot the Kata VM with the desired CPU count.  Most software works, possibly with a loss of performance, but mssql does not start.

xref https://github.com/kata-containers/kata-containers/issues/3793
xref https://github.com/microsoft/mssql-docker/issues/162

Caveats (at present):

1) Annotations in benchmark-operator workloads do not appear to be functioning correctly at present; I'm attempting to determine whether that's due to my error or a problem in benchmark-operator.

2) CI will need an update to set KATA_CPUOFFLINE_WORKAROUND=True

### CODE REVIEW

File: benchmark_runner/main/environment_variables.py [hunk 1]
```diff
@@ -37,6 +37,10 @@ def __init__(self):
         self._environment_variables_dict['elasticsearch_user'] = os.environ.get('ELASTICSEARCH_USER', '')
         self._environment_variables_dict['elasticsearch_password'] = os.environ.get('ELASTICSEARCH_PASSWORD', '')
 
+        # Workaround for Kata CPU offline problem in 4.9/4.10
+        # Set to True to
+        self._environment_variables_dict['kata_cpuoffline_workaround'] = os.environ.get('KATA_CPUOFFLINE_WORKAROUND', '')
```
Comment (ebattat) [inline benchmark_runner/main/environment_variables.py:42]: Do you mean we need to enable it for all our workloads run ?

Comment (RobertKrawitz) [reply benchmark_runner/main/environment_variables.py:42]: We should enable it in the CI runs, and then when we have a proper fix, take it out.

Comment (ebattat) [reply benchmark_runner/main/environment_variables.py:42]: For all the workloads, right ?

**Comment by ebattat** [comment]
Need to enable the environment variables in the following Github actions workflows:
-e KATA_CPUOFFLINE_WORKAROUND='True'

`https://github.com/redhat-performance/benchmark-runner/blob/afad66aef21055916b329e3e24f2c0692f2c3729/.github/workflows/Nightly_Func_Env_CI.yml#L108`
Add it after:
-e RUN_TYPE='func_ci'

`https://github.com/redhat-performance/benchmark-runner/blob/afad66aef21055916b329e3e24f2c0692f2c3729/.github/workflows/Nightly_Func_Env_Operator_CI.yml#L282`
Add it after:
-e RUN_TYPE='func_ci'

`https://github.com/redhat-performance/benchmark-runner/blob/afad66aef21055916b329e3e24f2c0692f2c3729/.github/workflows/Nightly_Perf_Env_CI.yml#L140`
Add it after:
-e RUN_TYPE='perf_ci'

`https://github.com/redhat-performance/benchmark-runner/blob/afad66aef21055916b329e3e24f2c0692f2c3729/.github/workflows/Nightly_Perf_Env_Operator_CI.yml#L282`
Add it after:
-e RUN_TYPE='perf_ci'

Also need to rebase branch

**Comment by RobertKrawitz** [comment]
Yep, starting work on it.

**Comment by RobertKrawitz** [comment]
Rebased and updated.

**Review by ebattat — APPROVED** [review]
/LGTM

**Pull request merged**